### PR TITLE
update to support 'getPixelForTick'

### PR DESCRIPTION
**Pull request by njcaruso**
Great plugin.  Wasn't working for a line-chart, and determined it was due to my data, and you using "getPixelForValue" instead of "getPixelForTick".  So I added an option to `useTicks`.

### CODE REVIEW

**Comment by sifex** [comment]
This can also be applied to BoxAnnotation. Fixed something I've trying to fix for 3 weeks. 

Have you got a donation box @njcaruso? Love to buy you a coffee.

File: chartjs-plugin-annotation.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*!
  * chartjs-plugin-annotation.js
  * http://chartjs.org/
- * Version: 0.5.3
+ * Version: 0.5.4
```
Comment (compwright) [inline chartjs-plugin-annotation.js]: Please revert all changes to this file, we will build it during the release process.

File: chartjs-plugin-annotation.min.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 /*!
  * chartjs-plugin-annotation.js
  * http://chartjs.org/
- * Version: 0.5.3
```
Comment (compwright) [inline chartjs-plugin-annotation.min.js]: Please revert all changes to this file, we will build it during the release process.

File: package.json [hunk 3]
```diff
@@ -1,13 +1,13 @@
 {
   "name": "chartjs-plugin-annotation",
   "description": "Annotations for Chart.js",
-  "version": "0.5.3",
+  "version": "0.5.4",
```
Comment (compwright) [inline package.json]: Please revert all changes to this file, we will build it during the release process.

File: src/types/line.js [hunk 4]
```diff
@@ -27,8 +27,13 @@ module.exports = function(Chart) {
 			var scale = chartInstance.scales[options.scaleID];
 			var pixel, endPixel;
 			if (scale) {
-				pixel = helpers.isValid(options.value) ? scale.getPixelForValue(options.value) : NaN;
-				endPixel = helpers.isValid(options.endValue) ? scale.getPixelForValue(options.endValue) : pixel;
+				if (options.useTicks) {
```
Comment (compwright) [inline src/types/line.js]: We should auto-detect whether this is needed based on the scale configuration.

File: README.md [hunk 5]
```diff
@@ -66,6 +66,8 @@ Vertical or horizontal lines are supported.
 	// ID of the scale to bind onto
 	scaleID: 'y-axis-0',
 
+	useTicks: true, // optional, and helpful if using 'x-axis-0' with ticks for data and mode = 'vertical'
```
Comment (compwright) [inline README.md]: We shouldn't add a separate option for this. Let's auto-detect based on the scale configuration instead.

**Comment by compwright** [comment]
This may fix #3, #35, #47. This PR needs to be tested with other scale types, and if they are working now, add examples.

**Comment by njcaruso** [comment]
I did end up also doing this for the box.js, (code snippet below).  Sorry the code isn't in a state to update the PR.

Auto-detect is definitely the way to go.  Unfortunately I'm completely swapped right now, but may get to it in the next week.

```
			if (xScale) {
				if (options.useTicks) {
					min = helpers.isValid(options.xMin) ? xScale.getPixelForTick(options.xMin) : chartArea.left;
					max = helpers.isValid(options.xMax) ? xScale.getPixelForTick(options.xMax) : chartArea.right;
				} else {
					min = helpers.isValid(options.xMin) ? xScale.getPixelForValue(options.xMin) : chartArea.left;
					max = helpers.isValid(options.xMax) ? xScale.getPixelForValue(options.xMax) : chartArea.right;
				}
				left = Math.min(min, max);
				right = Math.max(min, max);
			}

```

**Comment by compwright** [comment]
@njcaruso Please review #50 and see if it resolves your issue. If not, we can re-open this and I can rebase and edit it if you don't have time to, but I will need a test case first in CodePen or JSFiddle showing the issue that this PR fixes. Thanks!

**Pull request closed without merging**